github.go

```diff
@@ -61,6 +62,7 @@ func NewGithubClient(
 	token string,
 	releaseFilter ReleaseFilter,
 	assetFilter AssetFilter,
+	connectionTimeout time.Duration,
 ) *GithubLocator {
 	var httpClient *http.Client
```

Comment:
Since we don't allow specifying the http client set the timeout directly?
```suggestion
	httpClient := &http.Client{Timeout: connectionTimeout}
```

Reply:
If we do this, we still have to set timeout like this

```go
httpClient.Timeout = connectionTimeout
```

for `oauth2 client here`

```go
httpClient = oauth2.NewClient(context.Background(), src)
```

wdyt ?

Reply:
Good point. So probably something like this is the best to also not effect the default http client.
```
	var httpClient *http.Client
	if token != "" {
		src := oauth2.StaticTokenSource(
			&oauth2.Token{AccessToken: token},
		)
		httpClient = oauth2.NewClient(context.Background(), src)
		httpClient.Timeout = connectionTimeout
	} else {
		httpClient = &http.Client{Timeout: connectionTimeout}
	}
```

Reply:
Thanks, I have updated the behaviour now.